### PR TITLE
Add Pod PID Limit in Kubelet Configuration

### LINKED ISSUE #3351: Add Pod PID Limit in Kubelet Configuration
### Description of the feature you would like to add / User story

As a kubeone user, I would like to configure the Pod PID Limit from `kubeone.yaml` to pass the CIS Benchmark test `4.2.13 Ensure that a limit is set on pod PIDs (Manual)`

### Solution details

This is an example of the Kubelet configuration

```yaml
hosts:
- kubelet:
    systemReserved: xxxxx
    kubeReserved: xxxxx
    evictionHard: xxxxx
    podPidsLimit: 10000

```

### Additional information

In case the podPidsLimit is not used. It would be great to use the default value used by kubelet

### PR DESCRIPTION
**Pull request by mohamed-rafraf**
**What this PR does / why we need it**:
This PR adds Pod PID Limit in Kubelet Configuration. We can Pass the CIS Benchmark `4.2.13 Ensure that a limit is set on pod PIDs (Manual)`
**Which issue(s) this PR fixes**:

Fixes #3351

**What type of PR is this?**

/kind api-change
/kind feature

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
The Kubeone HostConfig support Pod PID Limits in its Kubelet Config
```

**Documentation**:

```documentation
T.B.D
```

### CODE REVIEW

**Comment by mohamed-rafraf** [comment]
I built a Kubeone binary and created an AWS cluster to run CIS Benchmark:

image

File: pkg/apis/kubeone/v1beta2/types.go [hunk 1]
```diff
@@ -353,6 +353,11 @@ type KubeletConfig struct {
 	// If not provided, default value provided by kubelet will be used
 	// (max. 110 pods per node)
 	MaxPods *int32 `json:"maxPods,omitempty"`
+
+	// PodPidsLimit configures the maximum number of processes running in a Pod
+	// If not provided, default value provided by kubelet will be used -1
+	// See more about pid-limiting at: https://kubernetes.io/docs/concepts/policy/pid-limiting/
+	PodPidsLimit *int64 `json:"podPidsLimit,omitempty"`
```
Comment (kron4eg) [inline pkg/apis/kubeone/v1beta2/types.go:360]: Please don't change v1beta2. We want to get rid of it in the future.

File: pkg/apis/kubeone/types.go [hunk 2]
```diff
@@ -343,6 +343,11 @@ type KubeletConfig struct {
 	// If not provided, default value provided by kubelet will be used
 	// (max. 110 pods per node)
 	MaxPods *int32 `json:"maxPods,omitempty"`
+
+	// PodPidsLimit configures the maximum number of processes running in a Pod
```
Comment (xmudrii) [inline pkg/apis/kubeone/types.go:347]: There should be a dot at the end of this line for better API docs rendering, i.e. `Pod.`

File: pkg/apis/kubeone/types.go [hunk 3]
```diff
@@ -343,6 +343,11 @@ type KubeletConfig struct {
...
+	// If not provided, default value provided by kubelet will be used -1
```
Comment (xmudrii) [inline pkg/apis/kubeone/types.go:348]: Same here, we should add a dot at the end of this line.

File: pkg/templates/kubeadm/v1beta3/kubeadm.go [hunk 4]
```diff
@@ -590,6 +590,14 @@ func newNodeRegistration(s *state.State, host kubeoneapi.HostConfig) kubeadmv1be
 		kubeletCLIFlags["max-pods"] = strconv.Itoa(int(*m))
 	}
 
+	if m := host.Kubelet.PodPidsLimit; m != nil {
+		kubeletCLIFlags["pod-max-pids"] = strconv.Itoa(int(*m))
+	} else {
+		// Set default value if PodsPidsLimits is nil
+		// in order to pass the 4.2.13 Check in CIS Benchmark 1.8
+		kubeletCLIFlags["pod-max-pids"] = strconv.Itoa(-1)
+	}
```
Comment (xmudrii) [inline pkg/templates/kubeadm/v1beta3/kubeadm.go:599]: Have we checked if this solves the CIS benchmark issue or they require setting a value that's not -1?

Comment (mohamed-rafraf) [reply pkg/templates/kubeadm/v1beta3/kubeadm.go:599]: CIS Benchmark requires to put a value and don't rely on kubelet default value. So in our case if the pod pid limit we will put the default value of kubelet to pass the test

Comment (xmudrii) [reply pkg/templates/kubeadm/v1beta3/kubeadm.go:599]: That's great, thanks for following up on this!

**Pull request merged**